cmd/bbolt/command_check.go

```diff
@@ -4,25 +4,72 @@ import (
 	"fmt"
 
 	"github.com/spf13/cobra"
+	"github.com/spf13/pflag"
 
 	bolt "go.etcd.io/bbolt"
 	"go.etcd.io/bbolt/internal/guts_cli"
 )
 
 func newCheckCommand() *cobra.Command {
 	checkCmd := &cobra.Command{
-		Use:   "check <bbolt-file>",
-		Short: "verify integrity of bbolt database data",
+		Use:   "check <subcommand>",
+		Short: "check related commands",
+	}
+
+	checkCmd.AddCommand(newCheckDBCommand())
+	checkCmd.AddCommand(newCheckPageCommand())
+
+	return checkCmd
+}
+
+func newCheckDBCommand() *cobra.Command {
+	checkDBCmd := &cobra.Command{
+		Use:   "check-db <bbolt-file>",
+		Short: "Verifies integrity of whole bbolt database data",
 		Args:  cobra.ExactArgs(1),
 		RunE: func(cmd *cobra.Command, args []string) error {
-			return checkFunc(cmd, args[0])
+			return checkFunc(cmd, args[0], nil)
 		},
 	}
 
-	return checkCmd
+	return checkDBCmd
+}
+
+type checkPageOption struct {
+	pageId uint64
 }
 
-func checkFunc(cmd *cobra.Command, dbPath string) error {
+func (o *checkPageOption) AddFlags(fs *pflag.FlagSet) {
+	fs.Uint64VarP(&o.pageId, "pageId", "", o.pageId, "page Id")
+	_ = cobra.MarkFlagRequired(fs, "pageId")
+}
```

Comment:
If this is the only argument (and required) for `check-page`, would it make sense to make it a required parameter instead? i.e. `check page <bbolt-file> <page-id>`?

On the other hand, if we kept `check` as a single command (without subcommands), then keeping the `pageId` argument makes sense.

Reply:
cc @ahrtr how should we proceed ?

Reply:
Yes, let's add a flag `--pageId` for the `bbolt check` command instead of adding a new sub command.

Reply:
@ahrtr i think having a sub command is more appropriate .. it is the same design as in `surgery` cmd 

wdyt ?

Reply:
- We are just enhancing the `bbolt check` command, we just add an optional flag/parameter to guide the check where to start check the database.
- Also there is only one method `(*Tx) Check`, which accepts options, including the pageId. It makes little sense to map one method to two commands. https://github.com/etcd-io/bbolt/blob/a718144dd11f9e0fb770592d5a5f217b639711c4/tx_check.go#L21
- Each surgery command is for a different functionality.

Reply:
updated .. ptal 👍🏽